### ISSUE TITLE
directline errors

### ISSUE THREAD

**Issue by ubreddy** [Microsoft/BotBuilder#5266]
I keep getting directline errors once a while multiple times during the day....

e.g.
getaddrinfo ENOTFOUND directline.botframework.com directline.botframework.com:443

and
some errors like below
https://directline.botframework.com/v3/conversations/I02GjBASIPz4DOmNQeTxKL/activities/I02GjBASIPz4DOmNQeTxKL%7C0000000' failed: [503] Service Unavailable

https://directline.botframework.com/v3/conversations/6lg94qXdQbv6WOaUTf42YO/activities/6lg94qXdQbv6WOaUTf42YO%7C0000000' failed: [500] Internal Server Error

https://directline.botframework.com/v3/conversations/HBVeZlhbpVx1B1cjKD5QXS/activities/HBVeZlhbpVx1B1cjKD5QXS%7C0000000' failed: [502] Bad Gateway
Any reason why ? it stops our webchat conversation flow....

Appid: 3b2bea4a-c6ac-4e54-a926-cbee6a1c1d2a

**Comment by vincec-msft** [comment]
The errors on the 1st and 2nd requests were caused by problems with Direct Line talking to internal storage services. We are improving the retry logic on those calls.

The 3rd request never got to Direct Line. Something on the bot or between it and Direct Line caused the error.

There will be transient errors such as these. I recommend retry logic with appropriate backoff times so that your bot doesn't get throttled. It would be nice if the SDK had such logic built in but I'm not sure if it does.

Regarding the |0000000s: this is the activity id that the bot is responding to. It looks like every post from your bot uses the same activity id. That's a little strange but not a problem. Especially if your bot is sending proactive messages, then it would only have the initial message to respond to so this pattern would be expected.

When posting reports please include the exact time of the errors. Thanks.

**Comment by ubreddy** [comment]
yes, activity ID ending with 0000000 are proactive messages.... 
If the conversation is closed and sending proactive messages to the same conversation would result in any of these above errors?
It should not and the directline and SDK should handle that state gracefully.
Strangely these are not getting logged in Application insights too....

**Comment by vincec-msft** [comment]
The errors around 14:12 were caused by a bug in a Direct Line dependency. We are working with that team to resolve the issue. It is a high priority for all teams involved.
